sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/SynchronousAccessor.java

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.implementation;
+
+import java.time.Duration;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Supplier;
+
+/**
+ * Synchronizes reactor threads accessing/instantiating a common value {@code T}.
+ *
+ * @param <T> The value being instantiated / accessed.
+ */
+public class SynchronousAccessor<T> {
+    private T cache;
```

Comment:
Does `cache` need to be `volatile` in this case? This is for synchronizing multiple threads and volatile will enforce the object state to be up to date when used in different threads.